### DISCUSSION TITLE
How to use AssetLoader.GetAssets() ?

### DISCUSSION THREAD

**Discussion by LaurentInSeattle** [AvaloniaUI/Avalonia#15744] [Q&A]
Hi all, 

I am building a localization model based on Tim's sample code. 
I am trying to list all resource files present in a subfolder of the regular "Assets" folder to detect language files.
I was thinking that GetAssets would do just that but it always returns an empty enumerable.

        string uriString = string.Format("avares:{0}/{1}", AssetsFolder, LanguagesFolder);
        var translations = AssetLoader.GetAssets(new Uri(uriString), null).ToList();

Accessing the language files does work with: 
            string uriString = string.Format("avares:{0}/{1}/{2}.axaml", AssetsFolder, LanguagesFolder, targetLanguage);
            var uri = new Uri(uriString);
            var newLanguage = new ResourceInclude(uri) { Source = uri };

So I think the URI's are correct.

Full Source: https://github.com/LaurentInSeattle/TextoCopier/blob/main/TextoCopier/TextoCopier/Models/LocalizerModel.cs 

On Windows 10, Visual Studio 2022, .Net 8 with Ava 11.0.10

**Comment by stevemonaco** [comment]
Try including the assembly name in the Uri and see what happens. I don't know how flexible the underlying implementation is, but you can certainly query for all assets in an assembly and filter later. I use this approach: https://github.com/stevemonaco/AvaloniaDemos/blob/master/Debugging/DebugPlus/DebugPlus.cs#L38

**Reply by LaurentInSeattle** [reply]
The assembly name was already in the URI. And I just double checked..

**Reply by stevemonaco** [reply]
You're also missing the `//` in `avares://`. At least in what you posted.

**Reply by LaurentInSeattle** [reply]
The final uri string is containing the slashes..

**Reply by LaurentInSeattle** [reply]
Specifying only the assembly name only returns the avalonia-logo.ico generated by the Visual Studio template

**Reply by timunie** [reply]
@LaurentInSeattle I suggest to create a minimal sample to play around. Probably you can also attach it here if that also doesn't work.
